packages/web-runtime/tests/unit/container/sse/shares.spec.ts

```diff
@@ -0,0 +1,770 @@
+      })
+      const space = mock<SpaceResource>({ id: sharedResource.storageId })
+      const mocks = getMocks({ currentRouteFilesSpacesGeneric: true, spaces: [space] })
+      const sseData = mock<EventSchemaType>({
+        itemid: sharedResource.id,
+        spaceid: sharedResource.storageId,
+        parentitemid: sharedResource.parentFolderId,
+        initiatorid: 'local1'
+      })
+      mocks.clientService.webdav.getFileInfo.mockResolvedValue(
+        mock<Resource>({ ...sharedResource })
+      )
+      await onSSELinkRemovedEvent({ sseData, ...mocks })
+      expect(mocks.clientService.webdav.getFileInfo).not.toHaveBeenCalled()
+      expect(mocks.resourcesStore.upsertResource).not.toHaveBeenCalled()
+      expect(mocks.resourcesStore.updateResourceField).not.toHaveBeenCalled()
+      expect(mocks.resourcesStore.removeResources).not.not.toHaveBeenCalled()
+    })
+  })
+})
+
+const getMocks = ({
+  currentFolder = mockDeep<Resource>({
+    id: 'currenFolder!currentFolder',
+    isFolder: true,
+    storageId: 'space1'
+  }),
+  resources = [],
+  spaces = [mockDeep<SpaceResource>({ id: 'space1' })],
+  currentRouteFilesSpacesGeneric = false,
+  currentRouteFilesSpacesProjects = false,
+  currentRouteFilesSharesViaLink = false,
+  currentRouteFilesSharesWithMe = false,
+  currentRouteFilesSharesWithOthers = false
+}: {
+  currentFolder?: Resource
+  resources?: Resource[]
+  spaces?: SpaceResource[]
+  currentRouteFilesSpacesGeneric?: boolean
+  currentRouteFilesSpacesProjects?: boolean
+  currentRouteFilesSharesViaLink?: boolean
+  currentRouteFilesSharesWithMe?: boolean
+  currentRouteFilesSharesWithOthers?: boolean
+} = {}) => {
+  createTestingPinia()
+  const resourcesStore = useResourcesStore()
+  resourcesStore.currentFolder = currentFolder
+  resourcesStore.resources = resources
+  const spacesStore = useSpacesStore()
+  spacesStore.spaces = spaces
+  const messageStore = useMessages()
+  const userStore = useUserStore()
+  userStore.user = mockDeep<User>({ id: '1' })
+  const sharesStore = useSharesStore()
+  const clientService = mockDeep<ClientService>({ initiatorId: 'local1' })
+  const previewService = mockDeep<PreviewService>()
+
+  const language = mockDeep<Language>({
+    $gettext: vi.fn((m) => m)
+  })
+  const resourceQueue = mockDeep<PQueue>()
+
+  let currentRoute = createLocationSpaces('files-spaces-generic')
+  if (currentRouteFilesSpacesGeneric) {
+    currentRoute = createLocationSpaces('files-spaces-generic')
+  }
+  if (currentRouteFilesSpacesProjects) {
+    currentRoute = createLocationSpaces('files-spaces-projects')
+  }
+  if (currentRouteFilesSharesViaLink) {
+    currentRoute = createLocationShares('files-shares-via-link')
+  }
+  if (currentRouteFilesSharesWithMe) {
+    currentRoute = createLocationShares('files-shares-with-me')
+  }
+  if (currentRouteFilesSharesWithOthers) {
+    currentRoute = createLocationShares('files-shares-with-others')
+  }
```

Comment:
This should then fix the type issues:

```suggestion
  let routeName: string
  if (currentRouteFilesSpacesGeneric) {
    routeName = 'files-spaces-generic'
  }
  if (currentRouteFilesSpacesProjects) {
    routeName = 'files-spaces-projects'
  }
  if (currentRouteFilesSharesViaLink) {
    routeName = 'files-shares-via-link'
  }
  if (currentRouteFilesSharesWithMe) {
    routeName = 'files-shares-with-me'
  }
  if (currentRouteFilesSharesWithOthers) {
    routeName = 'files-shares-with-others'
  }

  const currentRoute = mock<RouteLocation>({ name: routeName })
```